packages/compass-workspace/src/components/workspace-tabs.tsx

```diff
@@ -0,0 +1,330 @@
+  ({ tabs, onSelect, onClose, selectedTabIndex }: SortableListProps) => (
+    <div className={sortableItemContainerStyles}>
+      {tabs.map((tab: TabProps, index: number) => (
+        <SortableItem
+          key={`tab-${index}-${tab.subtitle}`}
+          // `index` is used internally by the SortableContainer hoc,
+          // so we pass our own `tabIndex`.
+          index={index}
+          tabIndex={index}
+          tab={tab}
+          onSelect={onSelect}
+          onClose={onClose}
+          selectedTabIndex={selectedTabIndex}
+        />
+      ))}
+    </div>
+  )
+);
+
+type WorkspaceTabsProps = {
+  darkMode?: boolean;
+  onCreateNewTab: () => void;
+  onSelectTab: (tabIndex: number) => void;
+  onCloseTab: (tabIndex: number) => void;
+  onMoveTab: (oldTabIndex: number, newTabIndex: number) => void;
+  tabs: TabProps[];
+  selectedTabIndex: number;
+};
+
+type TabProps = {
+  subtitle: string;
+  tabContentId: string;
+  title: string;
+  renderIcon: (
+    iconProps: Partial<React.ComponentProps<typeof Icon>>
+  ) => JSX.Element;
+};
+
+export function useRovingTabIndex<T extends HTMLElement = HTMLElement>({
+  currentTabbable,
+}: {
+  currentTabbable: number;
+}): React.HTMLProps<T> {
+  const rootNode = useRef<T | null>(null);
+  const [focusProps, focusState] = useFocusState();
+
+  const focusTabbable = useCallback(() => {
+    const selector = `[role="tab"]:nth-child(${
+      currentTabbable + 1 /* nth child starts at 1. */
+    })`;
+    rootNode.current?.querySelector<T>(selector)?.focus();
+  }, [rootNode, currentTabbable]);
+
+  useEffect(() => {
+    if (
+      [
+        FocusState.Focus,
+        FocusState.FocusVisible,
+        FocusState.FocusWithin,
+        FocusState.FocusWithinVisible,
+      ].includes(focusState)
+    ) {
+      focusTabbable();
+    }
+  }, [focusState, focusTabbable]);
+
+  return { ref: rootNode, ...focusProps };
+}
+
+function UnthemedWorkspaceTabs({
+  darkMode,
+  onCreateNewTab,
+  onCloseTab,
+  onMoveTab,
+  onSelectTab,
+  tabs,
+  selectedTabIndex,
+}: WorkspaceTabsProps) {
+  const rovingFocusProps = useRovingTabIndex<HTMLDivElement>({
+    currentTabbable: selectedTabIndex,
+  });
+  const tabContainerRef = useRef<HTMLDivElement>(null);
+
+  const [navigationProps] = useTabListKeyboardNavigation<HTMLDivElement>({
+    selectedTabIndex,
+    onSelectTab,
+    tabsCount: tabs.length,
+  });
+
+  const tabContainerProps = mergeProps<HTMLDivElement>(
+    rovingFocusProps,
+    navigationProps
+  );
+
+  const onSortEnd = useCallback(
+    ({ oldIndex, newIndex }) => {
+      onMoveTab(oldIndex, newIndex);
+    },
+    [onMoveTab]
+  );
+
+  return (
+    <div
+      className={cx(
+        tabsContainerStyles,
+        darkMode ? tabsContainerDarkStyles : tabsContainerLightStyles
+      )}
+    >
+      <div className={tabsListContainerStyles}>
+        <div
+          className={tabsListStyles}
+          role="tablist"
+          aria-label="Workspace Tabs"
```

Comment:
This should be probably also be a prop for now and say something like "Collections" or "Collection Workspaces" for now, does this makes sense?